extension_templates/forecasting.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+"""
```

Comment:
This is nice, could this go into the extension guidelines?

Reply:
I think it should be linked at the top of - the file is meant to be completely self-explanatory, accompanied by the more mathematical conceptual model.

Reply:
I'm not going to change the docs though in this PR, that would be dependent on this PR in order to keep feature additions well-delineated.

Reply:
The root directory doesn't seem like the right place for it. Why don't we move this to a new PR for doc changes?

Reply:
Why isn't the root directory the right place? The extender user doesn't want to look into the nested folder structure of sktime. They just want to get a template for the file they need to write, and it should work plug&play with sktime. I'd advocate to have one folder which collects all extension templates. The extension guidelines tell people how to build a local estimator for extension. Contributing back to sktime is an advanced case of that, but separate.

Reply:
I see this as documentation and that should live in the root directory, but somewhere in the docs, no?

Reply:
it's not documentation, it's not an md or an rst file, it's a py file that the extender would fill in.
Putting it in the documentation folder would "hide" it and make it less useful. I think it is its own "kind" of file, like the examples which also deserves its own folder.

The idea is:

* have extension guidelines in the docs
* the extension guidelines give a description and point to the template file
* the user then takes the template file, copies it in the intended location (sktime or own repo) and starts filling it out, following instructions in the file and the extension guidelines

If the file is "hidden", this workflow is unnecessarily complicated.

Reply:
I like the idea, but prefer not to add the templates to the root directory, especially since there are more templates to come. Why don't we create a `templates/` folder?

Reply:
slightly confused, it already is? It's in a folder called `extension_templates`

Reply:
Ah yes, though the file itself was in the root directory, would prefer `templates` as it's shorter, but either way it's ok.

Reply:
but `templates` is less descriptive :smiley: - I'm not too set on this though.

Reply:
We should mention them in the README too and perhaps add a short README to the folder itself too but let's do that in a separate PR

Reply:
either name works for now

Reply:
ideally we also import them into the sphinx rst doc files somehow and have some basic tests on them to make sure they are correct

Reply:
yes, agreed - my proposed steps for that future PR are:

* take the current markdown for forecaster specs and extension templates
* simplify, since the interface has simplified; re-work with reference to the py template
* move to sphinx rst, integrate to current docs